packages/salesforcedx-lightning-lsp-common/src/indexer/tagInfo.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Location } from 'vscode-languageserver';
+import { ClassMember } from '../decorators';
+import { AttributeInfo } from './attributeInfo';
+
+export type TagType = 'STANDARD' | 'SYSTEM' | 'CUSTOM';
+
+// Type definition for TagInfo data structure
+export type TagInfo = {
+    file: string | null;
+    type: TagType;
+    lwc: boolean;
+    attributes: AttributeInfo[];
+    location?: Location;
+    documentation?: string;
+    name?: string;
+    namespace?: string;
+    properties?: ClassMember[];
+    methods?: ClassMember[];
+};
+
+// Factory function to create TagInfo objects
+export const createTagInfo = (
```

Comment:
this one does "do a default" but you could do that by making `documentation` required (so people have to pass in an empty string), or (if we control all consumers) having them handle there not being a tag.

otherwise, it's not doing much